### PR TITLE
[#9] [Backend] As a user, I can sign in with email and password

### LINKED ISSUE #9: [Backend] As a user, I can sign in with email and password
## Why
The user can log in with their credentials.

## Acceptance Criteria
The API endpoint `/api/v1/oauth/token` is called with the following body:
```JSON
{
    "grant_type": "password",
    "email": "{{email}}",
    "password": "{{password}}",
    "client_id": "{{client_id}}",
    "client_secret": "{{client_secret}}"
}
```
Unit tests are added and passed.

## Resources
The API documentation is available on this publicly accessible [Postman collection](https://documenter.getpostman.com/view/11835486/UVsPNPfU).

### PR DESCRIPTION
**Pull request by nkhanh44**
- Close #9 

## What happened 👀

- Updated env and Secret with new CLIENT_ID and CLIENT_SECRET.
- Removed api_service and add the authentication_api_service.
- Added login_model, login_request, and login_response for handling requests and responses from calling API
- Added unit tests for authentication repository and login use case.

## Insight 📝

- Used [japx](https://pub.dev/packages/japx) for parsing JSON:API.
- In LoginResponse, we have the toLoginModel to convert that response to LoginModel for storing it later.

## Proof Of Work 📹

Unit tests for `authentication_repository_test` and `login_use_case` all passed

### CODE REVIEW

File: test/api/repositories/authentication_repository_test.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    test('When login successfully, it returns correct model', () async {
+      final loginResponse = LoginResponse(
+        id: "",
+        accessToken: "",
+        tokenType: "",
+        expiresIn: 0,
+        refreshToken: "",
+        createdAt: 0,
+      );
```
Comment (Thieurom) [inline test/api/repositories/authentication_repository_test.dart:38]: This is optional 🙏 Is it possible to put this instance creation as a static method for `LoginResponse` so we can use it in other locations 🤔 
Something like:
```dart
final loginResponse = LoginResponse.dummy()
```

File: test/api/repositories/authentication_repository_test.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey_flutter/repositories/authentication_repository.dart';
+import 'package:survey_flutter/api/exception/network_exceptions.dart';
+import 'package:survey_flutter/model/response/login_response.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('AuthenticationRepositoryTest', () {
```
Comment (Thieurom) [inline test/api/repositories/authentication_repository_test.dart:11]: Do we have a convention to name the `group` here, which the class name, or class name with the `Test` suffix will be used 🙏

Comment (nkhanh44) [reply test/api/repositories/authentication_repository_test.dart:11]: AFAIK, we don't have a convention to name the group, AuthenticationRepositoryTest is just a snake case of authentication_repository_test 🙏

File: test/api/repositories/authentication_repository_test.dart (hunk 1 above)
Comment (nkhanh44) [reply test/api/repositories/authentication_repository_test.dart]: Good suggestion 👍 , I updated all at https://github.com/nimblehq/flutter-ic-khanh-thieu/pull/51/commits/dff6a2549abeb02e72cf50686bbfbdc14ffaba4f 🙏

**Review by Thieurom — APPROVED** [review]
Great work 🙏

File: lib/api/response_decoder.dart [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import 'package:japx/japx.dart';
+
+Map<String, dynamic> decode(Map<String, dynamic> json) {
```
Comment (doannimble) [inline lib/api/response_decoder.dart:3]: I think we should wrap it into a class name ResponseDecodeder 🤔 
```suggestion
class ResponseDecoder {

static Map<String, dynamic> decode(Map<String, dynamic> json) {
```

**Review by doannimble — COMMENTED** [review]
Good work, only one tiny concern, rest lgtm

File: lib/api/response_decoder.dart (hunk 3 above)
Comment (nkhanh44) [reply lib/api/response_decoder.dart]: Thank you, I updated it at https://github.com/nimblehq/flutter-ic-khanh-thieu/pull/51/commits/7504b5f344ff3c450706889d59b1614edef186dc 🙌

**Review by doannimble — APPROVED** [review]
lgtm now 🏃

**Pull request merged**